### PR TITLE
Add AddDynamicPageFeedExample

### PR DESCRIPTION
**Pull request by devchas**

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+import com.google.protobuf.Int64Value;
+import com.google.protobuf.StringValue;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: Please add class javadoc

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param campaignId the campaign ID.
+   * @param adGroupId the ad group ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long campaignId, long adGroupId) {
+    String dsaPageUrlLabel = "discounts";
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: A line comment indicating what this is for would be helpful

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+      return feedDetails;
+    }
+  }
+
+  /**
+   * Creates a feed mapping for a given feed.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID.
+   * @param feedDetails a MAP of the names and IDs of feed attributes.
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: s/MAP/map

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+      Map<String, Long> feedDetails,
+      String feedResourceName,
+      String dsaPageUrlLabel) {
+    List<String> urls =
+        ImmutableList.of(
+            "http://www.example.com/discounts/rental-cars",
+            "http://www.example.com/discounts/hotel-deals",
+            "http://www.example.com/discounts/flight-deals");
+
+    // Creates a label attribute.
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: nit: "Creates a value for the label attribute."

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+    // Creates the service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues the search request.
+      SearchPagedResponse response = googleAdsServiceClient.search(request);
+
+      // Throws an exception if the campaign is not a DSA campaign.
+      // The server will return an error if the campaign is not a DSA campaign, but this
+      // exception is easier to interpret.
+      if (!response
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: Formatting looks strange

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+            .build();
+
+    // Creates the service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues the search request.
+      SearchPagedResponse response = googleAdsServiceClient.search(request);
+
+      // Throws an exception if the campaign is not a DSA campaign.
+      // The server will return an error if the campaign is not a DSA campaign, but this
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: This comment's a bit confusing. Is it meant to explain that the server will return an error _later_ if we try to set the DSA setting?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 7]
```diff
@@ -0,0 +1,516 @@
...
+          .getResponse()
+          .getResults(0)
+          .getCampaign()
+          .hasDynamicSearchAdsSetting()) {
+        throw new IllegalArgumentException(
+            String.format("Campaign with ID '%s' is not a " + "DSA campaign.", campaignId));
+      }
+
+      // Retrieves and returns the DSA setting.
+      return response
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: formatting

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java [hunk 8]
```diff
@@ -0,0 +1,516 @@
...
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID.
+   * @param adGroupId the ad group ID.
+   * @param dsaPageUrlLabel the label for the DSA page URLs.
+   */
+  private static void addDsaTarget(
+      GoogleAdsClient googleAdsClient, long customerId, long adGroupId, String dsaPageUrlLabel) {
+    String adGroupResourceName = ResourceNames.adGroup(customerId, adGroupId);
+
+    // Creates the webpage condition info.
```
Comment (jradcliff) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: The comments here explain what but not why. Could you make them more informative so they cover more than just the type of each object?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java (hunk 1 above)
Comment (BenRKarl) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: You can copy the language used in the [Python example](https://github.com/googleads/google-ads-python/blob/master/examples/advanced_operations/add_dynamic_page_feed.py#L15) if it's helpful

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java (hunk 5 above)
Comment (devchas) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: That's correct according to google-java-format.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java (hunk 6 above)
Comment (devchas) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: I've updated the language to make this more clear. I could remove this. When testing the error case, it's a bit easier to figure out what's happening when you throw the exception.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java (hunk 7 above)
Comment (devchas) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java]: Correct according to google-java-format.

**Pull request merged**